lib/ui/painting.dart

```diff
@@ -13,19 +13,19 @@ part of dart.ui;
 // would cause a crash. It is, however, acceptable for error messages to be much
 // less useful or correct in release mode than in debug mode.
 //
-// Painting APIs will also warn about arguments representing NaN or infinite
-// coordinates, which can not be rendered by Skia.
+// Painting APIs will also warn about arguments representing NaN coordinates,
+// which can not be rendered by Skia.
 
 bool _rectIsValid(Rect rect) {
-  return rect != null && rect.isFinite;
+  return rect != null && !rect.left.isNaN && !rect.top.isNaN && !rect.right.isNaN && !rect.bottom.isNaN;
 }
 
 bool _rrectIsValid(RRect rrect) {
-  return rrect != null && rrect.isFinite;
+  return rrect != null && !rrect.left.isNaN && !rrect.top.isNaN && !rrect.right.isNaN && !rrect.bottom.isNaN;
```

Comment:
there's also all the radii

Maybe for rect and rrect we should just do something like `rrect != null && !rrect._value.any((double value) => value.isNaN)`